### PR TITLE
Bugfix: Internallogger creates folder, even when turned off.

### LINKED ISSUE #1229: Directory created even if internalLogLevel set to off?
I noticed that the directory _c:\temp\_ of value `internalLogFile="c:\temp\nlog-internal.log"` is created even when `internalLogLevel` is set to `"Off"`:
```
internalLogLevel="Off"
internalLogFile="C:\temp\nlog-internal.log"
```

Is this on purpose or should the directory _c:\temp\_ only be created when the `internalLogLevel` is not set to `"Off"`? It's a minor issue but now I have an empty _temp_ directory and I didn't know what caused it at first..

Tested with NLog version: 4.2.3.

### PR DESCRIPTION
**Pull request by eduardorascon**
Fixes #1229 

When LogLevel is set to "Off" NLog should not create an emtpy folder or try to write a log file to a folder that doesn´t exists.

### CODE REVIEW

File: tests/NLog.UnitTests/Common/InternalLoggerTests.cs [hunk 1]
```diff
@@ -418,22 +416,15 @@ public void CreateDirectoriesIfNeededTests()
             // Set the log file, which will only create the needed directories
             InternalLogger.LogFile = tempFile;
 
-            Assert.True(Directory.Exists(randomSubDirectory));
+            Assert.True(Directory.Exists(randomSubDirectory) == shouldCreateDirectory);
```
Comment (304NotModified) [inline tests/NLog.UnitTests/Common/InternalLoggerTests.cs]: Please use assert.equal here

File: tests/NLog.UnitTests/Common/InternalLoggerTests.cs [hunk 2]
```diff
@@ -38,6 +38,7 @@
 using Xunit;
 using NLog.Common;
 using System.Text;
+using Xunit.Extensions;
```
Comment (304NotModified) [inline tests/NLog.UnitTests/Common/InternalLoggerTests.cs]: I think we need `#if !SILVERLIGHT` here and at the test.

**Comment by eduardorascon** [comment]
Already made those changes. Should I create another pull request?

**Comment by 304NotModified** [comment]
No that's not needed

File: tests/NLog.UnitTests/Common/InternalLoggerTests.cs (hunk 2 above)
Comment (eduardorascon) [inline tests/NLog.UnitTests/Common/InternalLoggerTests.cs]: CreateDirectoriesIfNeededTests is already inside an `#if !SILVERLIGHT` at [line 174](https://github.com/eduardorascon/NLog/blob/fix-create-directories/tests/NLog.UnitTests/Common/InternalLoggerTests.cs#L174).

**Comment by eduardorascon** [comment]
@304NotModified Can you give me some tips on how to fix this appveyor problem?

**Comment by 304NotModified** [comment]
I think it's fine now. One multi threaded test is lately unstable. See #1237

**Comment by 304NotModified** [comment]
Thanks!

**Comment by eduardorascon** [comment]
Thanks you, you were very helpful.

**Pull request merged**